packages/cli/src/controller/generate-controller.ts

```diff
@@ -221,65 +260,121 @@ function filterExistingFragments<T extends Fragment | ConstructorFragment>(
   return cleanFragments;
 }
 
-export function filterExistingMethods(
+export type ManifestExtractor<T> = (
+  dataSources: T,
+  casedInputAddress: string | undefined
+) => {
+  existingEvents: string[];
+  existingFunctions: string[];
+};
+
+export function filterExistingMethods<T>(
   eventFragments: Record<string, EventFragment>,
   functionFragments: Record<string, FunctionFragment>,
-  dataSources: EthereumDs[],
-  address: string | undefined
+  dataSources: T,
+  address: string | undefined,
+  extractor: ManifestExtractor<T>
 ): [Record<string, EventFragment>, Record<string, FunctionFragment>] {
+  const {existingEvents, existingFunctions} = extractor(dataSources, address && address.toLowerCase());
+
+  return [
+    filterExistingFragments<EventFragment>(eventFragments, existingEvents),
+    filterExistingFragments<FunctionFragment>(functionFragments, existingFunctions),
+  ];
+}
+
+export const yamlExtractor: ManifestExtractor<EthereumDs[]> = (dataSources, casedInputAddress) => {
   const existingEvents: string[] = [];
   const existingFunctions: string[] = [];
 
-  const casedInputAddress = address && address.toLowerCase();
-
   dataSources
-    .filter((d) => {
+    .filter((d: EthereumDs) => {
       if (casedInputAddress && d.options.address) {
-        return casedInputAddress.toLowerCase() === d.options.address.toLowerCase();
+        return casedInputAddress === d.options.address.toLowerCase();
       }
       return casedInputAddress === d.options.address || (!casedInputAddress && !d.options.address);
     })
-    .forEach((ds) => {
-      ds.mapping.handlers.forEach((handler) => {
-        // if (casedInputAddress !== ds.options.address) return;
-        if ('topics' in handler.filter) {
-          // topic[0] is the method
-          existingEvents.push((handler.filter as EthereumLogFilter).topics[0]);
-        }
-        if ('function' in handler.filter) {
-          existingFunctions.push((handler.filter as EthereumTransactionFilter).function);
-        }
+    .forEach((handler: any) => {
+      if ('topics' in handler.filter) {
+        existingEvents.push((handler.filter as EthereumLogFilter).topics[0]);
+      }
+      if ('function' in handler.filter) {
+        existingFunctions.push((handler.filter as EthereumTransactionFilter).function);
+      }
+    });
+
+  return {existingEvents, existingFunctions};
+};
+
+export const tsExtractor: ManifestExtractor<string> = (dataSources, casedInputAddress) => {
+  const existingEvents: string[] = [];
+  const existingFunctions: string[] = [];
+
+  splitArrayString(dataSources)
+    .filter((d) => {
+      const match = d.match(ADDRESS_REG);
+      return match && match[1].toLowerCase() === casedInputAddress;
```

Comment:
I think this check needs to be added.
```suggestion
      return match && match.length >= 2 && match[1].toLowerCase() === casedInputAddress;
```